### PR TITLE
ref(symbolicator): Replace log with tracing

### PR DESCRIPTION
**Pull request by loewenheim**
As a first step towards using `tracing` in symbolicator, this simply replaces the various `log` macros with their `tracing` equivalents. The more difficult part is setting up the subscribers (`tracing`'s loggers):

* The `Pretty`, `Simplified`, and `Json` log formats use the builtin styles `pretty`, `compact`, and `Json`, respectively. Notably, the `LogRecord` struct is obsolete. The same goes for `BreadcrumbLogger`, whose functionality is now handled by  `sentry-tracing`.
* Unfortunately, `tracing`'s `LevelFilter` doesn't implement `Deserialize`, so `config.rs` contains a manual deserialization function.

This PR doesn't exactly play to `tracing`'s strengths, as there are no spans and no structured logs. It's intended to introduce `tracing` with minimum hassle.

### CODE REVIEW

File: crates/symbolicator/src/config.rs [hunk 1]
```diff
@@ -366,6 +368,44 @@ impl Config {
     }
 }
 
+#[derive(Debug)]
+struct LevelFilterVisitor;
```
Comment (Swatinem) [inline crates/symbolicator/src/config.rs:372]: an alternative would be to define our own enum for this with a derived `Deserialize`, and convert between these two directly.

**Review by Swatinem — COMMENTED** [review]
This is actually pretty straight forward, I like it!

Maybe you can take a look at #515 and cherry-pick the instrumentation and sentry-tracing layer.
Other than that, please reach out to ops about whether we actually use these logs (json formatted?) somewhere. If we might change the format or if we need to keep backwards compatibility.

**Comment by loewenheim** [comment]
> Maybe you can take a look at #515 and cherry-pick the instrumentation and sentry-tracing layer.

I'd rather add instrumentation and structure in a second PR tbh. As for the sentry-tracing layer, is there anything more that needs to be done other than add a `SentryLayer` to the subscriber?

**Comment by loewenheim** [comment]
So before the most recent commit, some python tests were failing with a message about how they couldn't find a port or some such. Turns out, these tests start a symbolicator process and then wait for a specific log line from `actix_web` from which they parse the port that symbolicator is listening on. And since I hadn't set up `log` integration, the `actix_web` logs were silently discarded and the tests never found out the port 😅

**Comment by flub** [comment]
> Other than that, please reach out to ops about whether we actually use these logs (json formatted?) somewhere. If we might change the format or if we need to keep backwards compatibility.

Yes we use JSON formatted logs in production, you can get to them from the symbolicator metrics dashboard by following the "stackdriver logs" link.

**Comment by loewenheim** [comment]
The most recent commit (https://github.com/getsentry/symbolicator/pull/534/commits/5142030efc4a2170e6f7382196ec322552f2008c) solves the problem of keeping log output exactly the same in the simplest way: instead of using a `tracing::Subscriber`, we keep the logger and convert all `tracing` events to `log` records. This has the advantage of being very easy (really just add the `tracing` dependency with the `log` feature enabled and replace all the `log` invocations), but the disadvantage that it doesn't get rid of log entirely. Moreover, once we want to use `tracing` in more sophisticated ways, we will have to use a subscriber anyway. Thoughts?

**Comment by flub** [comment]
I preferred the version where you removed log entirely.  We should be able to make sure the tracing subscriber json output includes at least the same fields as we had before and then we're good.

I think in the future it would be nice to also start using the tracing macros with extra fields which would show up in the json log and make our log search much more useful.

**Comment by loewenheim** [comment]
@flub Is it clear that there's no harm in adding extra fields to the json logs?

**Comment by flub** [comment]
> @flub Is it clear that there's no harm in adding extra fields to the json logs?

I'm like 95% sure... 😄   @oioki might be able to give a better answer?

**Comment by loewenheim** [comment]
Alright, I think this is finally ready for serious consideration. JSON log output doesn't line up exactly, but after talking to @oioki we think it shouldn't be a problem.
Old log line:
```
{"timestamp":"2022-01-25T12:44:44.202043Z","level":"INFO","logger":"symbolicator::server","message":"Starting server on 127.0.0.1:3021","module_path":"symbolicator::server","filename":"crates/symbolicator/src/server.rs","lineno":43}
```
New log line:
```
{"timestamp":"2022-01-25T12:52:19.015333Z","level":"INFO","message":"Starting server on 127.0.0.1:3021","target":"symbolicator::server","filename":"crates/symbolicator/src/server.rs","line_number":43}
```

**Review by Swatinem — APPROVED** [review]

**Review by flub — APPROVED** [review]

**Comment by flub** [comment]
agree the json differences are fine.  let's merge this 🎉

**Pull request merged**